### ISSUE TITLE
Neural Networks: Back Propgation

### ISSUE THREAD

**Issue by sophryu99** [sophryu99/TIL#29]
<img width="288" alt="Screen Shot 2021-09-28 at 7 15 04 PM" src="https://user-images.githubusercontent.com/46921003/135191649-5aa961ac-61b3-42b1-ba0a-1fcaabec20eb.png">

- Algorithm for training feedforward neural networks: iteratively improving the prediction of the model by updating the mode's weights based on prediction error(loss)
- Instead of forward matrix calculation, going backwards doing **partial differentiation**, figuring how the input chain has to change to equal the desired output. 
  - Repeat this for every layer until the loss is backpropagated
  - The loss gets pushed into the weights of the models in a backward pass. 

- Computes the gradient of the loss function with respect to the weights of the network by the chain rule
